core/src/main/java/org/elasticsearch/index/mapper/Mapper.java

```diff
@@ -77,7 +76,17 @@ public String name() {
         }
 
         /** Returns a newly built mapper. */
-        public abstract Y build(BuilderContext context);
+        public final Y build(BuilderContext context) {
+            if (context.indexCreatedVersion() != null && context.indexCreatedVersion().onOrAfter(Version.V_5_0_0_alpha1)) {
```

Comment:
Because BuilderContext.indexCreatedVersion() has Nullable annotation...
Should we remove Nullable annotation in Mapper.BuilderContext?